### PR TITLE
Add support for ForexClient

### PR DESCRIPTION
**Pull request by Armando-Ibarra**

### CODE REVIEW

File: src/main/java/com/checkout/forex/four/QuoteResponse.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @SerializedName("source_amount")
+    private Long sourceAmount;
+
+    @SerializedName("destination_currency")
+    private Currency destinationCurrency;
+
+    @SerializedName("destination_amount")
+    private Long destinationAmount;
+
+    private Number rate;
```
Comment (rsaestrela) [inline src/main/java/com/checkout/forex/four/QuoteResponse.java:26]: What about making it a Double?

Comment (Armando-Ibarra) [reply src/main/java/com/checkout/forex/four/QuoteResponse.java:26]: I thought about that too, but in the project I found that we are using more a Number object

Comment (rsaestrela) [reply src/main/java/com/checkout/forex/four/QuoteResponse.java:26]: I was only able to find an usage on `Number` in the dispute module and actually it seems to be wrong. Number is an abstract class and apparently Gson does us the favor of instantiating a `LazilyParserNumber`. Please change to `Double` and do the same on `Dispute` and `DisputeDetailsResponse`.

File: documentation/docs/four_api/Forex.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Request an exchange rate between a source and destination currency pair that will be used to process one or more payouts. You must submit a payout with the FX quote identifier before the quote expires. If the FX quote identifier is omitted from a payout, and the source and destination currencies differ, the current market exchange rate will be used.
+
+```java
+QuoteRequest request = QuoteRequest.builder()
+        .sourceCurrency(Currency.GBP)
+        .sourceAmount(30000L)
+        .destinationCurrency(Currency.USD)
+        .processChannelId("pc_abcdefghijklmnopqrstuvwxyz")
+        .build();
```
Comment (rsaestrela) [inline documentation/docs/four_api/Forex.md:18]: Please remove the data from the example, leaving it like:

```
QuoteRequest request = QuoteRequest.builder()
        .sourceCurrency()
        .sourceAmount()
        .destinationCurrency()
        .processChannelId()
        .build();
```

**Pull request merged**